vllm/v1/engine/output_processor.py

```diff
@@ -17,6 +17,39 @@
                                    RequestStateStats)
 
 
+class RequestOutputCollector:
+    """Collects streamed RequestOutputs."""
+
+    def __init__(self, output_kind: RequestOutputKind):
+        self.aggregate = output_kind == RequestOutputKind.DELTA
+        self.output: Optional[RequestOutput] = None
+        self.ready = asyncio.Event()
+
+    def put(self, output: RequestOutput) -> None:
+        if self.output is None:
+            self.output = output
+            self.ready.set()
+        elif self.aggregate:
+            # Coalesce the outputs in delta case.
+            self.output.add(output)
+        else:
+            self.output = output
+
+    async def get(self) -> RequestOutput:
```

Comment:
Do you think we should have an invariant that `output` is not None if `self.ready.wait()` is true?